addons/website/models/ir_ui_view.py

```diff
@@ -29,6 +29,16 @@ def _compute_first_page_id(self):
         for view in self:
             view.first_page_id = self.env['website.page'].search([('view_id', '=', view.id)], limit=1)
 
+    @api.multi
+    def name_get(self):
+        res = []
+        for view in self:
+            view_name = view.name
+            if self._context.get('display_website') and view.website_id:
```

Comment:
move `if self._context.get('display_website')` before the loop to avoid useless iteration if not set (in most of the case)

Reply:
I can't since it is not calling super?
I have to loop to add to res

Reply:
call super in this case.

Reply:
if nothing to do, use default name_get of model
Else do mine.

And it make more sense for other module that would make the same